demo/src/Header.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Box, Link, Stack, Typography } from "@mui/material";
+import image from "../../assets/logo_square.png";
+
+export default function Header() {
+    return (
+        <header className="container">
+            <Stack direction={"row"}>
```

Comment:
I believe you can use `justifyContent="space-between"` here to ensure the logo appears on the right. (see https://mui.com/material-ui/react-stack/#interactive-demo)